src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -3423,11 +3425,46 @@ protected virtual void MarkMethodBody (MethodBody body)
 			MarkReflectionLikeDependencies (body, requiresReflectionMethodBodyScanner);
 		}
 
+		bool CheckRequiresReflectionMethodBodyScanner (MethodBody body)
+		{
+			if (ReflectionMethodBodyScanner.RequiresReflectionMethodBodyScannerForMethodBody (Context, body.Method))
```

Comment:
Could you please add a comment why we don't need a cache for this?
On the face value this is expensive - but I assume we don't care about the cache that much since it's only going to be called for reflection accessed methods (which should be rare).